pkg/datastore/datastore.go

```diff
@@ -0,0 +1,112 @@
+package datastore
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+)
+
+// Get gets a fqField from the datastore.
+func Get(ctx context.Context, addr string, key string, value interface{}) error {
+	reqBody, err := getRequestBody(key)
+	if err != nil {
+		return fmt.Errorf("building request body: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", addr+"/internal/datastore/reader/get", strings.NewReader(reqBody))
+	if err != nil {
+		return fmt.Errorf("creating request to datastore: %w", err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("sending request to datastore: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			body = []byte("[can not read body]")
+		}
+		return fmt.Errorf("got response `%s`: %s", resp.Status, body)
+	}
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("reading response body: %w", err)
+	}
+
+	var respData map[string]json.RawMessage
+	if err := json.Unmarshal(respBody, &respData); err != nil {
+		return fmt.Errorf("decoding response body `%s`: %w", respBody, err)
+	}
+
+	for field := range respData {
```

Comment:
This loop is hard to understand. Maybe first check that respData has length 1. If not, return error. Then check the one and only item and return error or nil if the value can be populated correctly.

Reply:
I don't know any other way to get the only value out of a map. Maybe we have to ask the datastore people to add a sane api-call to get one fqfield out of the datastore.